airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -114,6 +114,12 @@ class EmrContainerHook(AwsBaseHook):
         "CANCEL_PENDING",
     )
     SUCCESS_STATES = ("COMPLETED",)
+    TERMINAL_STATES = (
+        "COMPLETED",
+        "FAILED",
+        "CANCELLED",
+        "CANCEL_PENDING",
```

Comment:
Is "CANCEL_PENDING" really a terminal state?

Looks good to me other than this question.

Reply:
"CANCEL_PENDING" is being treated as a [failure state](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/emr.py#L111-L115) for the EMR container sensor, which is why I included it as a terminal state. Should it be removed?

Reply:
I don't know the EMR behavior well enough to say.  If you feel it's intentional and the right thing to do, then leave it in.  Just asking because the name strikes me as an intermediate state.

Reply:
Going by the code [here](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/emr.py#L111-L115) and [here](https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/sensors/test_emr_containers.py#L69-L72), I'll leave it in as a terminal state.